File: java/src/main/java/shapes/base/models/Rectangle.java
package shapes.base.models;

import shapes.base.commons.Dimensional;

public final class Rectangle implements Dimensional {
    private double width;
    private double height;

    public Rectangle(double width, double height) {
        double dimOne = (width > 0) ? width : 1;
        double dimTwo = (height > 0) ? height : 1;
        this.width = dimOne;
        this.height = dimTwo;
    }

    @Override
    public double area() {
        return width * height;
    }

    @Override
    public double perimeter() {
        return 2 * (width + height);
    }
}

File: kotlin/src/main/kotlin/shapes/base/models/Rectangle.kt
package shapes.base.models

import shapes.base.commons.Dimensional

class Rectangle : Dimensional {
    private var width: Double
    private var height: Double

    constructor(width: Double, height: Double) {
        val dimOne: Double = if (width > 0) width else 1.0
        val dimTwo: Double = if (height > 0) height else 1.0
        this.width = dimOne
        this.height = dimTwo
    }

    override fun area(): Double {
        return width * height
    }

    override fun perimeter(): Double {
        return 2 * (width + height)
    }
}
